Remove semver: prefix from instrumentation version (#3798)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

### Added

- Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)

> ### Changed

Reviewer: ```suggestion

### Changed
```

> ### Changed

Reviewer: ```suggestion

### Changed
```

### Added

- Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)

> - Remove the `SemVersion` function (#3681)

Reviewer: 1. This should be in "Removed" section (not added)
2. We should add in "Changed" something like: `semver: ` prefix is removed from version of the instrumentation scope.

Also

```suggestion
- Remove the `SemVersion` functions in instrumentation packages. (#3681)
```

Author: Just done, thanks your comment @pellared

Reviewer: Point 2 has not been addressed.

Author: Please checkout again

### Added

- Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)

> ### Changed
> - Remove `semver:` prefix from instrumentation version (#3681)
> - Deprecated `SemVersion` function

Reviewer: ```suggestion

### Changed

- Remove `semver:` prefix from instrumentation version (#3681)

### Deprecated

- `SemVersion` functions in instrumentation packages are deprecated, use `Version` instead. (#3681)
````

> ### Changed
> - Remove `semver:` prefix from instrumentation version (#3681)
> - Deprecated `SemVersion` function

Reviewer: ```suggestion

### Changed

- Remove `semver:` prefix from instrumentation version (#3681)

### Deprecated

- `SemVersion` functions in instrumentation packages are deprecated, use `Version` instead. (#3681)
````

- Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)

### Changed

> - Remove `semver:` prefix from instrumentation version. (#3681)
> 
> ### Deprecated
> 
> - `SemVersion` functions in instrumentation packages are deprecated, use `Version` instead. (#3681)

Reviewer 2: ```suggestion
- Remove `semver:` prefix from instrumentation version. (#3681, #3798)

### Deprecated

- `SemVersion` functions in instrumentation packages are deprecated, use `Version` instead. (#3681, #3798)
```
